src/arfsdao.ts

```diff
@@ -750,74 +810,391 @@ export class ArFSPrivateDrive extends ArFSEntity implements ArFSDriveEntity {
+			entity.txId,
+			entity.unixTime,
+			entity.parentFolderId,
+			entity.entityId
+		);
+		this.cipher = entity.cipher;
+		this.cipherIV = entity.cipherIV;
+		this.path = `${hierarchy.pathToFolderId(entity.parentFolderId)}${entity.name}`;
+		this.txIdPath = `${hierarchy.txPathToFolderId(entity.parentFolderId)}${entity.txId}`;
+		this.entityIdPath = `${hierarchy.entityPathToFolderId(entity.parentFolderId)}${entity.entityId}`;
+	}
+}
+
+export class ArFSPublicFile extends ArFSFileOrFolderEntity {
+	constructor(
+		readonly appName: string,
+		readonly appVersion: string,
+		readonly arFS: string,
+		readonly contentType: ContentType,
+		readonly driveId: DriveID,
+		readonly entityType: EntityType,
+		readonly name: string,
+		readonly txId: string,
+		readonly unixTime: number,
+		readonly parentFolderId: FolderID,
+		readonly fileId: FileID
+	) {
+		super(
+			appName,
+			appVersion,
+			arFS,
+			contentType,
+			driveId,
+			entityType,
+			name,
+			txId,
+			unixTime,
+			parentFolderId,
+			fileId
+		);
+	}
+}
+
+export class ArFSPrivateFile extends ArFSFileOrFolderEntity {
+	constructor(
+		readonly appName: string,
+		readonly appVersion: string,
+		readonly arFS: string,
+		readonly contentType: ContentType,
+		readonly driveId: DriveID,
+		readonly entityType: EntityType,
+		readonly name: string,
+		readonly txId: TransactionID,
+		readonly unixTime: number,
+		readonly parentFolderId: FolderID,
+		readonly fileId: FileID,
+		readonly cipher: string,
+		readonly cipherIV: string
+	) {
+		super(
+			appName,
+			appVersion,
+			arFS,
+			contentType,
+			driveId,
+			entityType,
+			name,
+			txId,
+			unixTime,
+			parentFolderId,
+			fileId
+		);
+	}
+}
+
+export class ArFSPublicFolder extends ArFSFileOrFolderEntity {
+	constructor(
+		readonly appName: string,
+		readonly appVersion: string,
+		readonly arFS: string,
+		readonly contentType: ContentType,
+		readonly driveId: DriveID,
+		readonly entityType: EntityType,
+		readonly name: string,
+		readonly txId: TransactionID,
+		readonly unixTime: number,
+		readonly parentFolderId: FolderID,
+		readonly entityId: FolderID
+	) {
+		super(
+			appName,
+			appVersion,
+			arFS,
+			contentType,
+			driveId,
+			entityType,
+			name,
+			txId,
+			unixTime,
+			parentFolderId,
+			entityId
+		);
+	}
+}
+export class ArFSPrivateFolder extends ArFSFileOrFolderEntity {
+	constructor(
+		readonly appName: string,
+		readonly appVersion: string,
+		readonly arFS: string,
+		readonly contentType: ContentType,
+		readonly driveId: DriveID,
+		readonly entityType: EntityType,
+		readonly name: string,
+		readonly txId: string,
+		readonly unixTime: number,
+		readonly parentFolderId: FolderID,
+		readonly entityId: FolderID,
+		readonly cipher: string,
+		readonly cipherIV: string
+	) {
+		super(
+			appName,
+			appVersion,
+			arFS,
+			contentType,
+			driveId,
+			entityType,
+			name,
+			txId,
+			unixTime,
+			parentFolderId,
+			entityId
+		);
+	}
+}
+export class FolderTreeNode {
```

Comment:
Lets move all the FolderHierarchy types into their own file. This source file is getting enormous 😬

Reply:
There: 4043255a613bb030d2c220d01a6a36729c03715c